main/ecrecover/FPSECP256K1/checkSqrtFpSecp256k1.zkasm

```diff
@@ -16,20 +16,20 @@
 
 ; Efficiency of the exponentiation:
 ; (p-1)/2 = 0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffff7ffffe17
-;             (*) 222 x 1, 1 x 0, 22 x 1, 4 x 0, 1 x 1, 1 x 0, 3 x 1
-;             (*) 222 x SM, S, 22 x SM, 4 x S, 1 x SM, 1 x S, 3 x SM    (S=square M=multiplication)
-;         BLOCK:  1         2  3        4      5       6      7
+;             (*) 222 x 1,  1 x 1, 22 x 1,  4 x 1, 1 x 1,  1 x 1, 3 x 1
+;             (*) 222 x SM, 1 x S, 22 x SM, 4 x S, 1 x SM, 1 x S, 3 x SM    (S=square M=multiplication)
+;         BLOCK:  1         2      3        4      5       6      7
 ;             (*) 222 + initial initialization
 
 VAR GLOBAL checkSqrtFpSecp256k1_base
 VAR GLOBAL checkSqrtFpSecp256k1_RR
 
 checkSqrtFpSecp256k1:
         ; RESOURCES:
-        ;   248 x (3 steps + 2 arith + 8 steps + 2 arith + 7 steps) = 992 ariths + 4464 steps
-        ;   6 x (1 step + 2 arith + 8 steps) = 12 ariths + 54 steps
+        ;   248 x (3 steps + 1 arith + 5 steps + 1 arith + 4 steps) = 496 ariths + 2976 steps
+        ;   6 x (1 step + 1 arith + 5 steps) = 6 ariths + 36 steps
         ;   6 steps + 1 binary
-        ; TOTAL = 1004 ariths + 1 binary + 4524 steps
+        ; TOTAL = 502 ariths + 1 binary + 3018 steps
```

Comment:
OPTIMIZATION, how now the operations are very short, unroll the operations or alternative, reduce at least setting of  SECP256K1_P, call and you can left result in best register for next operation.

Reply:
Done